src/internal/proc.js

```diff
@@ -420,6 +408,28 @@ export default function proc(
     cb.cancel = takeCb.cancel
   }
 
+  function runDeclarativeEffect(effect, effectId, cb) {
+    if (options.middleware) {
```

Comment:
what do you mean? this function is now taking care of all the declarative effects, if the middleware returns the same it should still handle the same logic.

The reason im doing this here and separating the declarative effects from the rest is that i dont want to pass the middleware arrays of effects or pending promises or internal stuff